src/storage/page_buffer.c

```diff
@@ -15911,16 +15922,29 @@ class pgbuf_page_flush_daemon_task : public cubthread::entry_task
 	  return;
 	}
 
-      /* did not timeout, someone requested flush... run at least once */
-      m_force_one_run = pgbuf_Page_flush_daemon->was_woken_up ();
+      refresh_page_flush_interval ();
+
+      bool force_one_run = false;
+      bool stop_iteration = false;
+
+      if (m_page_flush_interval_msec > 0)
```

Comment:
According to old code, the purpose is to force one run when page flush thread was waked. Are you sure this change is correct?

Reply:
Indeed, you are right. I was mislead by the return of `thread_daemon_timedwait` function, which returns `true` if daemon was not woken up and `false` if it was and then according to old code
```c
wakeup_interval = prm_get_integer_value (PRM_ID_PAGE_BG_FLUSH_INTERVAL_MSECS);
if (wakeup_interval > 0)
  {
    if (!thread_daemon_timedwait (&thread_Page_flush_thread, wakeup_interval))
      {
        /* did not timeout, someone requested flush... run at least once */
        force_one_run = true;          // A
      }
  }
else
  {
    thread_daemon_wait (&thread_Page_flush_thread);
    /* did not timeout, someone requested flush... run at least once */
    force_one_run = true;              // B
  }
```
in both cases **A** and **B** force_one_run is `true` daemon was woken up